### PR TITLE
adminExports: explode arrays of object in csv exports

### LINKED ISSUE #694: adminExport: arrays of object should be exploded in csv file
If an array in the responses contains objects (for example `scenarioData`: `[{id: 1, mode: 'car'}, {id: 2, mode: 'bus'}]`, the data in the csv is currently shown in one column named scenarioData, with `[Object]|[Object]`.

It should be exploded, as those arrays are usually of limited size and each interviews will have the same indices. In this case, there should be instead 4 columns: `scenarioData.0.id`, `scenarioData.0.mode`, `scenarioData.1.id`, `scenarioData.1.mode`.

### PR DESCRIPTION
**Pull request by tahini**
fixes #694

When a survey attribute is an array of complex types (object), instead of simply joining the values with `|`, we explode the paths to display each of its fields, ie `attribute.0.field1`, `attribute.0.field2`, `attribute.1.field1`, etc.

For arrays of simple type (strings, numbers, etc), the current behavior is kept.

### CODE REVIEW

**Review by kaligrafy — APPROVED** [review]

File: packages/evolution-backend/src/services/adminExport/exportToCsvByObjectWorkerPool.ts [hunk 1]
```diff
@@ -50,7 +50,26 @@ const getNestedAttributes = function (
             attributesAndObjectPaths.objectPaths.push(parentAttribute);
         }
         if (Array.isArray(_object[attribute])) {
-            attributesAndObjectPaths.arrayPaths.push((parentAttribute ? parentAttribute + '.' : '') + attributeRenamed);
+            // Is this an array of simple or complex types?
+            if (
+                _object[attribute].length > 0 &&
+                _object[attribute].every((obj) => typeof obj === 'object' && !Array.isArray(obj))
```
Comment (samuel-duhaime) [inline packages/evolution-backend/src/services/adminExport/exportToCsvByObjectWorkerPool.ts:56]: Can you name the if in a variable like something isArrayComplexTypes ?

File: packages/evolution-backend/src/services/adminExport/exportToCsvByObjectWorkerPool.ts [hunk 2]
```diff
@@ -75,6 +94,14 @@ const getPaths = function (parentPath, _object, attributes: string[] = []) {
         if (attribute === '_actions') {
             continue;
         }
+        if (
+            Array.isArray(_object[attribute]) &&
```
Comment (samuel-duhaime) [inline packages/evolution-backend/src/services/adminExport/exportToCsvByObjectWorkerPool.ts:98]: Same as the other stuff. Can you store that in a variable?

 Why do you need to check if it's an array and not on the other one? Isn't it supposed to be the same as the if statement on lines 55-56?

Comment (tahini) [reply packages/evolution-backend/src/services/adminExport/exportToCsvByObjectWorkerPool.ts]: Done, the if was extracted in a `isArrayOfObjects` function

**Review by samuel-duhaime — APPROVED** [review]
Okay. Good.

**Pull request merged**